compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/partition_cut.rs

```diff
@@ -12,36 +12,98 @@ use crate::optimization::dag::solo_key::symbolic_variance::SymbolicVariance;
 const ROUND_INNER_MULTI_PARAMETER: bool = false;
 const ROUND_EXTERNAL_MULTI_PARAMETER: bool = !ROUND_INNER_MULTI_PARAMETER && true;
 
+#[derive(Hash, Eq, PartialEq, Clone, Debug)]
+pub struct ExternalPartition {
+    pub name: String,
+    // TODO add params (maybe just macros)
+}
+
+impl std::fmt::Display for ExternalPartition {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{{ name: {} }}", self.name)?;
+        Ok(())
+    }
+}
+
 // TODO: keep both precisions
 // TODO: rounding lut should have its own partition based on max norm2 and precisions
 #[derive(Clone, Debug)]
 pub struct PartitionCut {
+    // TODO: add name to partitions
+
     // partition0 precision <= p_cut[0] < partition 1 precision <= p_cut[1] ...
     // precision are in the sens of Lut input precision and are sorted
     pub p_cut: Vec<(Precision, f64)>,
 
+    // Whether it has internal partitions or not
+    pub has_internal_partitions: bool,
+
```

Comment:
Reading below, I have a hard time understanding why this is needed. Why isn't it enough to check the length of `p_cut` ?

Reply:
the `test_tfhers_in_out_dot_compute` test showcase the issue. `p_cut` could actually be empty if there is no lut, and this was translated to having an internal partition. However, we wanted to ignore this default partition if change partition is used.

Reply:
p_cut empty used to mean 1 internal partition only. Each member of the vec introduce the limit of the partitiion, the last partition always accept all the remaining and is not part of the vec.
since in theory there could be circuit with just 1 external partition, we would be unable to say the p_cut object has only 1 partition.
if it's too confusive we can always add a last member in the p_cut fields explicitely with very high limits.